src/Engine/UnitActionLifecycles/CaptureLifecycle.java

```diff
@@ -116,8 +117,12 @@ public GameEventQueue getEvents(MapMaster map)
             {
               // Someone is losing their big, comfy chair.
               ArmyDefeatEvent defeat = new ArmyDefeatEvent(captureLocation.getOwner().army);
+              Commander newOwner = actor.CO;
+              // Capture territory for your primary CO in persistent tags
+              if( newOwner.gameRules.tagMode.supportsMultiCmdrSelect )
+                newOwner = newOwner.army.cos[0];
```

Comment:
Is the first CO in the array always the currently-active one (as in, we are shuffling them around in the array when tags happen)?
Even if we are doing that, it would be cleaner to have a `getActiveCO` or something rather than relying on outside classes directly accessing implicitly-meaningful array indices.

Reply:
Yes, for simplicity/consistency.
I suppose that's a fair point, though.

Reply:
When I went to actually write that code, I recalled why I didn't do it.
`getActiveCO() -> cos[0]` is useful for AWBW/persistent tags and harmless for single-CO... but team-merge doesn't have a "primary CO" concept.

Also, `getActiveCO()` is a lot more characters than `cos[0]`, which bothers me more than it should.

Reply:
> team-merge doesn't have a "primary CO" concept.

But then you'd still be using `cos[0]` anyway, yeah? I don't see how this change would have a negative impact (or change anything) even in that case.

> which bothers me more than it should

As a fan of verbose naming and clear intents, I agree that this should not bother you.
We all know that the real problem is not about variable length, but whether adjacent variables are justified properly. Fortunately, [there is help](https://github.com/timvieira/justified-variables).

Reply:
> But then you'd still be using `cos[0]` anyway, yeah?

Sure.
> I don't see how this change would have a negative impact (or change anything) even in that case.

Names have power, and "active/primary CO" gives a false sense of that concept always being concretely true.
> As a fan of verbose naming and clear intents, I agree that this should not bother you.

"Active", "primary", and "first" are all synonyms here anyway (for the cases you care about), so why have a problem with `cos[0]`?
> We all know that the real problem is not about variable length, but whether adjacent variables are justified properly. Fortunately, [there is help](https://github.com/timvieira/justified-variables).

Amazing. I'm not sure I wanted to know this exists.

Reply:
I might debate with you about those all being first (even "primary" and "first" are commonly used in different contexts), but it's more just that other code generally shouldn't need to know internal data structures of `Commander` to interact with it. It also just makes the intent more clear.

Reply:
Most of the codebase is interested in the internal data structures of the stuff that makes up our game state.
`GameInstance` to a lesser degree, but `Army`/map/terrain/units are all very fair game.
(And it's `Army.cos[0]`, `Commander` doesn't have a list of COs.)
> It also just makes the intent more clear.

I disagree, as noted above.

I think encapsulation for `Commander` makes a fair amount more sense (unless it's e.g. an AI that cares to know stuff like "am I a DoR CO?"), and to a lesser extent `UnitModel`.